include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -988,13 +988,13 @@ namespace VectorTools
 
 
 
-    template <int dim, typename VectorType, int fe_degree, int n_q_points_1d>
+    template <int dim, typename VectorType, int spacedim, int fe_degree, int n_q_points_1d>
     void project_parallel (const MatrixFree<dim,typename VectorType::value_type> &matrix_free,
                            const ConstraintMatrix &constraints,
                            const std_cxx11::function< VectorizedArray<typename VectorType::value_type> (const unsigned int, const unsigned int)> func,
                            VectorType &vec_result)
     {
-      const DoFHandler<dim> &dof = matrix_free.get_dof_handler();
+      const DoFHandler<dim,spacedim> &dof = matrix_free.get_dof_handler();
```

Comment:
You are right, it can't work right now. This is more a solution looking forward in case we extend that at some point. Not sure whether that is the best strategy but this would at least be the most consistent change with the other `project_parallel` function.